components/base/sensorcontrolled/velocities.go

```diff
@@ -36,85 +35,68 @@ func (sb *sensorBase) setupControlLoops() error {
 	return nil
 }
 
+func (sb *sensorBase) setupControlLoop(linear, angular control.PIDConfig) error {
+	// set the necessary options for a sensorcontrolled base
+	options := control.Options{
+		SensorFeedback2DVelocityControl: true,
+		LoopFrequency:                   20,
+		ControllableType:                "base_name",
+	}
+
+	// check if either linear or angular need to be tuned
+	if (linear.P == 0.0 && linear.I == 0.0 && linear.D == 0.0) ||
```

Comment:
[opt] we could make this check a method on PIDConfig. Would let us set the autoconfig flag by using 

`options.NeedsAutoTuning = linear.NeedsTuning || angular.NeedsTuning`

though theres definitely an argument to leave these checks as is so the conditions are explicit.

Reply:
hmm I'm not opposed, I'll try it out see if it makes sense

Reply:
need to test but I think what I added should work